homeassistant/components/sunweg/__init__.py

```diff
@@ -102,43 +101,49 @@ def get_api_value(
 
     def get_data(
         self,
-        entity_description: SunWEGSensorEntityDescription,
+        *,
+        api_variable_key: str,
+        api_variable_unit: str | None,
+        deep_name: str | None,
         device_type: DeviceType,
-        inverter_id: int = 0,
-        deep_name: str | None = None,
-    ) -> StateType | datetime.datetime:
+        inverter_id: int,
+        name: str | UndefinedType | None,
+        native_unit_of_measurement: str | None,
+        never_resets: bool,
+        previous_value_drop_threshold: float | None,
+    ) -> tuple[StateType | datetime.datetime, str | None]:
         """Get the data."""
         _LOGGER.debug(
             "Data request for: %s",
-            entity_description.name,
+            name,
         )
-        variable = entity_description.api_variable_key
-        previous_unit = entity_description.native_unit_of_measurement
+        variable = api_variable_key
+        previous_unit = native_unit_of_measurement
         api_value = self.get_api_value(variable, device_type, inverter_id, deep_name)
         previous_value = self.previous_values.get(variable)
         return_value = api_value
-        if entity_description.api_variable_unit is not None:
-            entity_description.native_unit_of_measurement = self.get_api_value(
-                entity_description.api_variable_unit,
```

Comment:
Is it possible keep this smaller and only do this change plus `return (return_value, native_unit_of_measurement)`?

Reply:
nevermind. I didn't see its going to get refactored above